drivers/api/no_os_i2c.c

```diff
@@ -147,7 +147,7 @@ void no_os_i2cbus_remove(uint32_t bus_number)
 
 		if (bus != NULL) {
 			no_os_free(bus);
-			bus = NULL;
+			i2c_table[bus_number] = NULL;
```

Comment:
It could be that the reason this doesn't behave as expected is that first, bus is not assigned correctly at line 140, it should be (not the & character):
```
&i2c_table[bus_number];
```

And on line 150 where you made the change, `*bus = NULL` should do.

Could you please try this ?